internal/jobs/launch_instance_gcp.go

```diff
@@ -83,5 +83,17 @@ func DoLaunchInstanceGCP(ctx context.Context, args *LaunchInstanceGCPTaskArgs) e
 		return fmt.Errorf("cannot update operation name for GCP : %w", err)
 	}
 
+	// For each instance that was created in GCP, add it as a DB record
+	for _, instanceId := range instances {
+		err = rDao.CreateInstance(ctx, &models.ReservationInstance{
+			ReservationID: args.ReservationID,
+			InstanceID:    *instanceId,
+		})
+		if err != nil {
+			return fmt.Errorf("cannot create instance reservation for id %d: %w", instanceId, err)
+		}
+		logger.Info().Str("instance_id", *instanceId).Msgf("Created new instance via GCP reservation %s", *opName)
+	}
```

Comment:
We need to do this in a DB transaction, I suggest to refactor the `CreateInstance` function to `CreateInstances` accepting slice of `ReservationInstance` and then using `dao.WithTransaction` to perform the operation atomically.

We can do this in a followup too, that is fine. We currently do this in AWS too.

Reply:
I think it is out of scope for this PR, no problem doing it as a follow up